applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -253,6 +327,350 @@ void NavierStokesWallCondition<3,3>::CalculateNormal(array_1d<double,3>& An )
+
+            BaseLHSMatrixGPcontribution(4,3) = rDataStruct.Normal[0] * N[0] * N[1] * wGauss;
+            BaseLHSMatrixGPcontribution(5,3) = rDataStruct.Normal[1] * N[0] * N[1] * wGauss;
+            BaseLHSMatrixGPcontribution(6,3) = rDataStruct.Normal[2] * N[0] * N[1] * wGauss;
+
+            BaseLHSMatrixGPcontribution(8,3) = rDataStruct.Normal[0] * N[0] * N[2] * wGauss;
+            BaseLHSMatrixGPcontribution(9,3) = rDataStruct.Normal[1] * N[0] * N[2] * wGauss;
+            BaseLHSMatrixGPcontribution(10,3)= rDataStruct.Normal[2] * N[0] * N[2] * wGauss;
+            // ----------------------------------------------------------------------------
+            BaseLHSMatrixGPcontribution(0,7) = rDataStruct.Normal[0] * N[1] * N[0] * wGauss;
+            BaseLHSMatrixGPcontribution(1,7) = rDataStruct.Normal[1] * N[1] * N[0] * wGauss;
+            BaseLHSMatrixGPcontribution(2,7) = rDataStruct.Normal[2] * N[1] * N[0] * wGauss;
+
+            BaseLHSMatrixGPcontribution(4,7) = rDataStruct.Normal[0] * N[1] * N[1] * wGauss;
+            BaseLHSMatrixGPcontribution(5,7) = rDataStruct.Normal[1] * N[1] * N[1] * wGauss;
+            BaseLHSMatrixGPcontribution(6,7) = rDataStruct.Normal[2] * N[1] * N[1] * wGauss;
+
+            BaseLHSMatrixGPcontribution(8,7) = rDataStruct.Normal[0] * N[1] * N[2] * wGauss;
+            BaseLHSMatrixGPcontribution(9,7) = rDataStruct.Normal[1] * N[1] * N[2] * wGauss;
+            BaseLHSMatrixGPcontribution(10,7)= rDataStruct.Normal[2] * N[1] * N[2] * wGauss;
+            // ----------------------------------------------------------------------------
+            BaseLHSMatrixGPcontribution(0,11) = rDataStruct.Normal[0] * N[2] * N[0] * wGauss;
+            BaseLHSMatrixGPcontribution(1,11) = rDataStruct.Normal[1] * N[2] * N[0] * wGauss;
+            BaseLHSMatrixGPcontribution(2,11) = rDataStruct.Normal[2] * N[2] * N[0] * wGauss;
+   
+            BaseLHSMatrixGPcontribution(4,11) = rDataStruct.Normal[0] * N[2] * N[1] * wGauss;
+            BaseLHSMatrixGPcontribution(5,11) = rDataStruct.Normal[1] * N[2] * N[1] * wGauss;
+            BaseLHSMatrixGPcontribution(6,11) = rDataStruct.Normal[2] * N[2] * N[1] * wGauss;
+   
+            BaseLHSMatrixGPcontribution(8,11) = rDataStruct.Normal[0] * N[2] * N[2] * wGauss;
+            BaseLHSMatrixGPcontribution(9,11) = rDataStruct.Normal[1] * N[2] * N[2] * wGauss;
+            BaseLHSMatrixGPcontribution(10,11)= rDataStruct.Normal[2] * N[2] * N[2] * wGauss;
+
+            BaseLHSMatrix += BaseLHSMatrixGPcontribution;
+        }
+    }
+
+    // KRATOS_WATCH( BaseLHSMatrix )
+
+    // Computation of NodalMultMatrix = ( [I] - (na)(na) ) for each node
+    std::vector<MatrixType> NodalMultMatrix(TNumNodes);
+    const MatrixType auxIdentMatrix = identity_matrix<double>(3);
+    MatrixType auxMatrix = zero_matrix<double>(3,3);
+    
+    for (SizeType nnode=0; nnode < TNumNodes; nnode++){
+        auxMatrix = prod( NodalNormals[nnode], trans(NodalNormals[nnode]) );
```

Comment:
I am sorry, but here I cannot follow:
The "auxMatrix" contains the (n_a) (n_a) part of the projection matrix [ I - (n_a) (n_a) ]. In my opinion, the normal of  the node should be included. What could be improved is to compute  [ I - (n_a) (n_a) ] in one statement and, thus, to eliminate "auxMatrix" (>> now implemented)
In case I am entirely misunderstanding your advise, it would be very kind if you could explain your thought once again, please.